homeassistant/components/syncthing/sensor.py

```diff
@@ -0,0 +1,264 @@
+"""Support for monitoring the Syncthing instance."""
+
+import logging
+
+import aiosyncthing
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.core import callback
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import (
+    DOMAIN,
+    FOLDER_PAUSED_RECEIVED,
+    FOLDER_SENSOR_ALERT_ICON,
+    FOLDER_SENSOR_DEFAULT_ICON,
+    FOLDER_SENSOR_ICONS,
+    FOLDER_SUMMARY_RECEIVED,
+    SCAN_INTERVAL,
+    SERVER_AVAILABLE,
+    SERVER_UNAVAILABLE,
+    STATE_CHANGED_RECEIVED,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Syncthing sensors."""
+    syncthing = hass.data[DOMAIN][config_entry.entry_id]
+
+    try:
+        config = await syncthing.system.config()
+        version = await syncthing.system.version()
+    except aiosyncthing.exceptions.SyncthingError as exception:
+        raise PlatformNotReady from exception
+
+    server_id = syncthing.server_id
+    entities = [
+        FolderSensor(
+            syncthing,
+            server_id,
+            folder["id"],
+            folder["label"],
+            version["version"],
+        )
+        for folder in config["folders"]
+    ]
+
+    async_add_entities(entities)
+
+
+class FolderSensor(SensorEntity):
+    """A Syncthing folder sensor."""
+
+    STATE_ATTRIBUTES = [
+        "errors",
+        "globalBytes",
```

Comment:
Normally we use snake_case names for the keys in the state attributes. Can we make this a dict instead that maps the service keys to the state attribute keys?